ewkcoffea/modules/objects_wwz.py

```diff
@@ -17,32 +17,50 @@ def get_cleaned_collection(obj_collection_a,obj_collection_b,drcut=0.4):
 ######### WWZ 4l analysis object selection #########
 
 # WWZ preselection for electrons
-def is_presel_wwz_ele(ele,tight):
+def is_presel_wwz_ele(ele,year,tight):
     mask = (
         (ele.pt               >  get_ec_param("wwz_pres_e_pt")) &
         (abs(ele.eta)         <  get_ec_param("wwz_pres_e_eta")) &
         (abs(ele.dxy)         <  get_ec_param("wwz_pres_e_dxy")) &
         (abs(ele.dz)          <  get_ec_param("wwz_pres_e_dz")) &
+        (ele.miniPFRelIso_all < get_ec_param("wwz_pres_e_miniPFRelIso_all")) &
         (abs(ele.sip3d)       <  get_ec_param("wwz_pres_e_sip3d")) &
-        (ele.miniPFRelIso_all <  get_ec_param("wwz_pres_e_miniPFRelIso_all")) &
         (ele.lostHits         <= get_ec_param("wwz_pres_e_lostHits"))
     )
-    if tight: mask = (mask & ele.convVeto & (ele.tightCharge == get_ec_param("wwz_pres_e_tightCharge")))
-    return mask
+    mask_2022 = (
+        (ele.pt               >  10.0) &
+        (abs(ele.eta)         <  2.5) &
+        (abs(ele.dxy)         <  0.05) &
+        (abs(ele.dz)          <  0.1)
+    )
+    if ("2022" in year): mask_return = (mask_2022 & ele.mvaIso_WP80)
+    if ("2022" not in year): mask_return = mask
+    if tight: mask_return = (mask_return & ele.convVeto & (ele.tightCharge == get_ec_param("wwz_pres_e_tightCharge")))
+    return mask_return
 
 
 # WWZ preselection for muons
-def is_presel_wwz_mu(mu):
+def is_presel_wwz_mu(mu,year):
     mask = (
         (mu.pt               >  get_ec_param("wwz_pres_m_pt")) &
         (abs(mu.eta)         <  get_ec_param("wwz_pres_m_eta")) &
         (abs(mu.dxy)         <  get_ec_param("wwz_pres_m_dxy")) &
+        (mu.miniPFRelIso_all < get_ec_param("wwz_pres_m_miniPFRelIso_all")) &
         (abs(mu.dz)          <  get_ec_param("wwz_pres_m_dz")) &
         (abs(mu.sip3d)       <  get_ec_param("wwz_pres_m_sip3d")) &
-        (mu.miniPFRelIso_all <  get_ec_param("wwz_pres_m_miniPFRelIso_all")) &
         (mu.mediumId)
     )
-    return mask
+    mask_2022 = (
+        (mu.pt               >  10.0) &
+        (abs(mu.eta)         <  2.4) &
+        (abs(mu.dxy)         <  0.05) &
+        (abs(mu.dz)          <  0.1) &
+        (mu.pfIsoId          >=  4) &
+        (mu.mediumId)
+    )
+    if ("2022" not in year): return_mask = mask
+    if ("2022" in year): return_mask = mask_2022
```

Comment:
Make this an else (and indent after the if blocks instead of doing it in line) to match the ele function.